File: LiveData/app/src/main/java/com/kurio/mvvmtesting/db/NoteRoomDatabase.java
package com.kurio.mvvmtesting.db;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;
import android.os.AsyncTask;
import android.support.annotation.NonNull;

import com.kurio.mvvmtesting.model.Note;

@Database(entities = {Note.class}, version = 1, exportSchema = false)
public abstract class NoteRoomDatabase extends RoomDatabase {
    public abstract NoteDao noteDao();

    private static NoteRoomDatabase instance;

    public static NoteRoomDatabase getInstance(Context mCtx) {

        if (instance == null) {
            instance = Room.databaseBuilder(mCtx, NoteRoomDatabase.class, "note_database")
                    .addCallback(roomDb)
                    .build();
        }
        return instance;
    }


    static RoomDatabase.Callback roomDb = new RoomDatabase.Callback() {
        @Override
        public void onCreate(@NonNull SupportSQLiteDatabase db) {
            super.onCreate(db);
//            new PopulateData(instance).execute();
        }
    };

    private static class PopulateData extends AsyncTask<Void, Void, Void> {
        NoteRoomDatabase noteRoomDatabase;

        PopulateData(NoteRoomDatabase noteRoomDatabase) {
            this.noteRoomDatabase = noteRoomDatabase;
        }

        @Override
        protected Void doInBackground(Void... voids) {
            for (int i = 0; i < 5; i++) {
                Note note = new Note("Title", "Description", 1);
                noteRoomDatabase.noteDao().insert(note);
            }

            return null;
        }
    }
}

File: LiveData (Kotlin)/app/src/main/java/com/kurio/mvvmtesting/db/NoteRoomDatabase.kt
package com.kurio.mvvmtesting.db

import android.arch.persistence.db.SupportSQLiteDatabase
import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context
import android.os.AsyncTask
import com.kurio.mvvmtesting.Constants

import com.kurio.mvvmtesting.model.Note

@Database(entities = [Note::class], version = 1, exportSchema = false)
abstract class NoteRoomDatabase : RoomDatabase() {
    abstract fun noteDao(): NoteDao

//    private class PopulateData internal constructor(internal var noteRoomDatabase: NoteRoomDatabase) : AsyncTask<Void, Void, Void>() {
//
//        override fun doInBackground(vararg voids: Void): Void? {
//            for (i in 0..4) {
//                val note = Note("Title", "Description", 1)
//                noteRoomDatabase.noteDao().insert(note)
//            }
//
//            return null
//        }
//    }

    companion object {

        private var instance: NoteRoomDatabase? = null

        fun getInstance(mCtx: Context): NoteRoomDatabase {

            if (instance == null) {
                instance = Room.databaseBuilder<NoteRoomDatabase>(mCtx, NoteRoomDatabase::class.java, Constants.DB_NAME)
                        .addCallback(roomDb)
                        .build()
            }
            return instance as NoteRoomDatabase
        }


        internal var roomDb: RoomDatabase.Callback = object : RoomDatabase.Callback() {
            override fun onCreate(db: SupportSQLiteDatabase) {
                super.onCreate(db)
                //            new PopulateData(instance).execute();
            }
        }
    }
}
